colossalai/gemini/ag_chunk.py

```diff
@@ -0,0 +1,365 @@
+import torch
+import torch.distributed as dist
+from typing import Optional, Dict
+
+from colossalai.utils import get_current_device
+from colossalai.gemini.chunk import TensorState, STATE_TRANS, TensorInfo, ChunkFullError, \
+    free_storage, alloc_storage
+
+
+class AgChunk:
+    def __init__(self,
+                 chunk_size: int,
+                 process_group: dist.ProcessGroup,

```

Comment:
It is torch pg

Reply:
Yes. I use torch pg here. I have explicate in below coments. Is there any problem?

Reply:
done